src/witan/httpapi/components/activities.clj

```diff
@@ -262,13 +285,28 @@
         (complete-receipt! db command-id nil)))))
 
 (defmethod on-event
-  [:kixi.datastore.metadatastore/sharing-change-rejected "1.0.0"]
+  [:kixi.datastore.metadatastore/update-rejected "1.0.0"]
   [db {:keys [kixi.comms.event/payload] :as event}]
   (let [command-id (:kixi.comms.command/id event)]
     (when-let [receipt (retreive-receipt db command-id)]
-      (let [file-id (get-in payload [:original :kixi.datastore.metadatastore/id])]
-        (create-error! db command-id file-id (-> payload :reason name))
-        (complete-receipt! db command-id (str "/api/files/" file-id "/errors/" command-id)))))  )
+      (if-let [file-id (get-metadatastore-id payload)]
+        (complete-receipt!
+         db
+         command-id
+         (create-error! db command-id file-id (-> payload :reason name)))
+        (log/error "Could't find a metadatastore ID in payload:" payload))))
+
+  (defmethod on-event
+    [:kixi.datastore.metadatastore/sharing-change-rejected "1.0.0"]
+    [db {:keys [kixi.comms.event/payload] :as event}]
+    (let [command-id (:kixi.comms.command/id event)]
+      (when-let [receipt (retreive-receipt db command-id)]
+        (if-let [file-id (get-metadatastore-id payload)]
+          (complete-receipt!
+           db
+           command-id
+           (create-error! db command-id file-id (-> payload :reason name)))
+          (log/error "Could't find a metadatastore ID in payload:" payload))))))
```

Comment:
should this throw an exception? (or should the exception happen in `get-metadatastore-id` as there should be an ID by this point.